packages/shared/src/components/modals/RanksModal.tsx

```diff
@@ -79,88 +25,183 @@ export interface RanksModalProps extends ModalProps {
-        className="self-start mt-4"
+  };
+
+  return (
+    <section className="overflow-hidden">
+      <ul
+        className="flex flex-nowrap"
+        style={{
+          transform: `translate(${RANK_OFFSET[showRank]})`,
+        }}
       >
-        Generate
-      </GoToDevCardButton>
-    );
-  }
+        {RANKS.map((_rank) => (
+          <li
+            className="flex relative flex-col justify-center items-center w-1/4"
+            key={_rank.name}
+          >
+            {_rank.level !== RANKS.length && (
+              <div
+                className={classNames(
+                  rankCompleted(showRank, _rank, progress)
+                    ? _rank.background
+                    : 'bg-white',
+                  'absolute w-3/4 h-px  top-[35%] left-[60%]',
+                )}
+              />
+            )}
+            <div
+              className={classNames(
+                'flex justify-center items-center relative bg-theme-bg-tertiary',
+                rankCompleted(showRank, _rank, progress) ? _rank.border : '',
+                showRank === _rank.level
+                  ? 'w-16 h-16'
+                  : 'w-10 h-10 border rounded-12',
+              )}
+            >
+              {showRank === _rank.level && (
+                <RadialProgress
+                  progress={progress}
+                  steps={finalRankCompleted ? 0 : _rank.steps}
+                  className={classNames(
+                    styles.radialProgress,
+                    'w-full h-full absolute inset-0',
+                  )}
+                  style={
+                    {
+                      '--radial-progress-completed-step': `var(--theme-rank-${
+                        finalRankCompleted ? showRank : showRank - 1
+                      }-color)`,
+                    } as CSSProperties
+                  }
+                />
+              )}
+              <Rank
+                rank={_rank.level}
+                className={classNames(
+                  rank === _rank.level ? 'w-10 h-10' : 'w-8 h-8',
+                )}
+                colorByRank={finalRankCompleted ? true : showRank > _rank.level}
+              />
+            </div>
+            <strong
+              className={classNames(
+                'mt-2',
+                rank === _rank.level
+                  ? 'typo-callout text-theme-status-success'
+                  : 'typo-footnote',
+                rankCompleted(showRank, _rank, progress)
+                  ? _rank.color
+                  : 'text-theme-label-tertiary',
+              )}
+            >
+              {_rank.name}
+            </strong>
+          </li>
+        ))}
+      </ul>
+      <p className="mt-1 mb-3 text-center text-theme-label-tertiary typo-footnote">
+        {getNextRankText({
+          nextRank: rank + 1,
+          rank,
+          finalRank: isFinalRank,
+          progress,
+          showNextLevel: false,
+        })}
+      </p>
+    </section>
+  );
```

Comment:
Looks somewhat lengthy. I think we can refactor a bit and split to helper components